sys-utils/renice.c

```diff
@@ -89,13 +95,18 @@ static int getprio(const int which, const int who, int *prio)
 	return 0;
 }
 
-static int donice(const int which, const int who, const int prio)
+static int donice(const int which, const int who, const int prio, const int relative)
 {
 	int oldprio, newprio;
 
 	if (getprio(which, who, &oldprio) != 0)
 		return 1;
-	if (setpriority(which, who, prio) < 0) {
+
+	newprio = oldprio;
+	if (relative)
+	    newprio += prio;
+
+	if (setpriority(which, who, newprio) < 0) {
```

Comment:
This is not correct. The code does not modify priority if the `relative` is not set (as you call setpriority() with newprio=oldprio). I guess we need the following:
```
if (relative)
   prior = oldprio + prio;

if (setpriority(which, who, prio) < 0)
```